### ISSUE TITLE
clang complains about missing parentheses in logical expression

### ISSUE THREAD

**Issue by H-M-H** [teeworlds/teeworlds#1476]
```
src/engine/client/serverbrowser.cpp:237:35: warning: '&&' within '||' [-Wlogical-op-parentheses]
        if(Type < 0 || Type >= NUM_TYPES && Type != m_ActServerlistType)
                    ~~ ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/engine/client/serverbrowser.cpp:237:35: note: place parentheses around the '&&' expression to silence this warning
        if(Type < 0 || Type >= NUM_TYPES && Type != m_ActServerlistType)
                                         ^
                       (                                               )
1 warning generated.
```

I am not sure what the desired behaviour is, so I just post this warning instead of a pullrequest trying to fix something.

**Comment by heinrich5991** [comment]
To me it looks like the desired expression is `|| Type == m_ActServerlistType`, i.e. also return early if we're already on the right page.

**Comment by H-M-H** [comment]
I guess your are right, at least this sounds more reasonable than the current code.

**Comment by oy** [comment]
fixed by 170b487430b93ea8ec3f46725d5955bddb0746dc
